File: backend/app/schemas/availability.py
from pydantic import BaseModel, Field
from typing import Dict, List, Optional
from datetime import date
import uuid

class TimeSlot(BaseModel):
    start: str
    end: str

class AvailabilityBase(BaseModel):
    timezone: str = "UTC"
    working_hours: Dict[str, List[TimeSlot]] = Field(default_factory=dict)
    meeting_duration: int = 30
    vacation_mode: bool = False
    vacation_start: Optional[date] = None
    vacation_end: Optional[date] = None

class AvailabilityUpdate(AvailabilityBase):
    pass

class AvailabilityResponse(AvailabilityBase):
    id: uuid.UUID
    user_id: uuid.UUID

    class Config:
        from_attributes = True

File: frontend/src/types/availability.ts
export interface TimeSlot {
  start: string;
  end: string;
}

export interface WorkingHours {
  [day: string]: TimeSlot[];
}

export interface UserAvailability {
  id: string;
  user_id: string;
  timezone: string;
  working_hours: WorkingHours;
  meeting_duration: number;
  vacation_mode: boolean;
  vacation_start: string | null;
  vacation_end: string | null;
}

export interface AvailabilityUpdate {
  timezone: string;
  working_hours: WorkingHours;
  meeting_duration: number;
  vacation_mode: boolean;
  vacation_start: string | null;
  vacation_end: string | null;
}

export const DAYS_OF_WEEK = [
  "monday",
  "tuesday",
  "wednesday",
  "thursday",
  "friday",
  "saturday",
  "sunday",
];
